src/olympia/addons/tests/test_views.py

```diff
@@ -1412,6 +1423,214 @@ class TestVersionViewSetCreateJWTAuth(TestVersionViewSetCreate):
+        request = APIRequestFactory().get('/')
+        request.version = 'v5'
+        request.user = self.user
+        assert data == VersionSerializer(
+            context={'request': request}
+        ).to_representation(version)
+
+    def test_not_authenticated(self):
+        self.client.logout_api()
+        response = self.client.patch(
+            self.url,
+            data={'release_notes': {'en-US': 'Something new'}},
+        )
+        assert response.status_code == 401
+        assert response.data == {
+            'detail': 'Authentication credentials were not provided.'
+        }
+        assert self.version.release_notes != 'Something new'
+
+    def test_not_your_addon(self):
+        self.addon.addonuser_set.get(user=self.user).update(
+            role=amo.AUTHOR_ROLE_DELETED
+        )
+        response = self.client.patch(
+            self.url,
+            data={'release_notes': {'en-US': 'Something new'}},
+        )
+        assert response.status_code == 403
+        assert response.data['detail'] == (
+            'You do not have permission to perform this action.'
+        )
+        assert self.version.release_notes != 'Something new'
+
+    def test_not_read_agreement(self):
+        self.user.update(read_dev_agreement=None)
+        response = self.client.patch(
+            self.url,
+            data={'release_notes': {'en-US': 'Something new'}},
+        )
+        assert response.status_code in [401, 403]  # JWT auth is a 401; web auth is 403
+        assert 'agreement' in response.data['detail'].lower()
+        assert self.version.release_notes != 'Something new'
+
+    def test_waffle_flag_disabled(self):
+        gates = {
+            'v5': (
+                gate
+                for gate in settings.DRF_API_GATES['v5']
+                if gate != 'addon-submission-api'
+            )
+        }
+        with override_settings(DRF_API_GATES=gates):
+            response = self.client.patch(
+                self.url,
+                data={'release_notes': {'en-US': 'Something new'}},
+            )
+        assert response.status_code == 403
+        assert response.data == {
+            'detail': 'You do not have permission to perform this action.'
+        }
+        assert self.version.release_notes != 'Something new'
+
+    def test_cant_update_upload(self):
+        self.version.update(version='123.b4')
+        upload = self.get_upload(
+            'webextension.xpi', user=self.user, source=amo.UPLOAD_SOURCE_ADDON_API
+        )
+        with mock.patch('olympia.addons.serializers.parse_addon') as parse_addon_mock:
+            response = self.client.patch(
+                self.url,
+                data={'upload': upload.uuid},
+            )
+            parse_addon_mock.assert_not_called()
+
+        assert response.status_code == 200, response.content
+        self.addon.reload()
+        self.version.reload()
+        assert self.version.version == '123.b4'
```

Comment:
I take it from this that if the upload was successful the version number would be something other than `123.b4`?